// @nodeidentity/common/build/core/api/ApiResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiResponse = exports.ResponseStatus = void 0;
var StatusCode;
(function (StatusCode) {
    StatusCode["SUCCESS"] = "10000";
    StatusCode["FAILURE"] = "10001";
    StatusCode["RETRY"] = "10002";
    StatusCode["INVALID_ACCESS_TOKEN"] = "10003";
})(StatusCode || (StatusCode = {}));
var ResponseStatus;
(function (ResponseStatus) {
    ResponseStatus[ResponseStatus["SUCCESS"] = 200] = "SUCCESS";
    ResponseStatus[ResponseStatus["BAD_REQUEST"] = 400] = "BAD_REQUEST";
    ResponseStatus[ResponseStatus["UNAUTHORIZED"] = 401] = "UNAUTHORIZED";
    ResponseStatus[ResponseStatus["FORBIDDEN"] = 403] = "FORBIDDEN";
    ResponseStatus[ResponseStatus["NOT_FOUND"] = 404] = "NOT_FOUND";
    ResponseStatus[ResponseStatus["INTERNAL_ERROR"] = 500] = "INTERNAL_ERROR";
})(ResponseStatus = exports.ResponseStatus || (exports.ResponseStatus = {}));
var ApiResponse = /** @class */ (function () {
    function ApiResponse(status, message, success, result) {
        this.status = status;
        this.message = message;
        this.success = success;
        this.result = result;
    }
    ApiResponse.prototype.prepare = function (res, response) {
        return res.status(this.status).json(ApiResponse.sanitize(response));
    };
    ApiResponse.prototype.send = function (res) {
        return this.prepare(res, this);
    };
    ApiResponse.sanitize = function (response) {
        var clone = {};
        Object.assign(clone, response);
        // delete {some_field};
        // delete clone.Status;
        for (var i in clone)
            if (typeof clone[i] === 'undefined')
                delete clone[i];
        return clone;
    };
    return ApiResponse;
}());
exports.ApiResponse = ApiResponse;
